frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx

```diff
@@ -0,0 +1,171 @@
+import './BillingUsageGraph.styles.scss';
+import '../../../lib/uPlotLib/uPlotLib.styles.scss';
+
+import { Color } from '@signozhq/design-tokens';
+import { Card, Flex, Typography } from 'antd';
+import { getComponentForPanelType } from 'constants/panelTypes';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { PropsTypePropsMap } from 'container/GridPanelSwitch/types';
+import { useIsDarkMode } from 'hooks/useDarkMode';
+import { useResizeObserver } from 'hooks/useDimensions';
+import tooltipPlugin from 'lib/uPlotLib/plugins/tooltipPlugin';
+import getAxes from 'lib/uPlotLib/utils/getAxes';
+import getRenderer from 'lib/uPlotLib/utils/getRenderer';
+import { getUPlotChartData } from 'lib/uPlotLib/utils/getUplotChartData';
+import { getXAxisScale } from 'lib/uPlotLib/utils/getXAxisScale';
+import { getYAxisScale } from 'lib/uPlotLib/utils/getYAxisScale';
+import { FC, useRef } from 'react';
+import uPlot from 'uplot';
+
+import {
+	convertDataToMetricRangePayload,
+	fillMissingValuesForQuantities,
+} from './utils';
+
+interface BillingUsageGraphProps {
+	data: any;
+	billAmount: number;
+}
+const paths = (
+	u: any,
+	seriesIdx: number,
+	idx0: number,
+	idx1: number,
+	extendGap: boolean,
+	buildClip: boolean,
+): uPlot.Series.PathBuilder => {
+	const s = u.series[seriesIdx];
+	const style = s.drawStyle;
+	const interp = s.lineInterpolation;
+
+	const renderer = getRenderer(style, interp);
+
+	return renderer(u, seriesIdx, idx0, idx1, extendGap, buildClip);
+};
+
+export function BillingUsageGraph(props: BillingUsageGraphProps): JSX.Element {
+	const { data, billAmount } = props;
+	const graphCompatibleData = convertDataToMetricRangePayload(data);
```

Comment:
memo this , as this is compute heavy and will be calculated on every re-render

Reply:
done